Source/WebKit/UIProcess/WebPageProxy.h

```diff
@@ -1573,6 +1574,8 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
     void didCancelForOpenPanel();
 
     WebPageCreationParameters creationParameters(WebProcessProxy&, DrawingAreaProxy&, RefPtr<API::WebsitePolicies>&& = nullptr);
```

Comment:
Instead of adding two more functions that each call creationParameters then modify the result, I was thinking that it would be better to just add some more parameters to creationParameters:

WebPageCreationParameters creationParameters(WebProcessProxy&, DrawingAreaProxy&, bool isProcessSwap, RefPtr<API::WebsitePolicies>&& = nullptr, std::optional<SubframeProcessPageParameters>&& = std::nullopt, etc.);

That way the function shows all of what the result depends on and returns a complete result instead of a caller wondering which function to call and what needs to be done with the result after calling the function.

Reply:
My rationale included:
- The new functions were clearly intended for prov/remote pages.
- The new functions only took whatever data was required.
- The new functions did return a complete result, ready for message-sending.
- The function calls were as short as possible.

Now we have one giant function declaration, where callers need to explicitly provide extra information (that was implied by the ...ForX function names), and to skip function parameters that are not relevant (but only in one case so far, so it's not too bad.)

Please have a look and confirm that that's what you prefer, thanks.

Note: I had to pass SubframeProcessPageParameters by raw pointer, to avoid having to `#include WebPageCreationParameters.h` in WebPageProxy.h; Is there a better way?
Otherwise if we were to add the include, we would be able to pass `std::optional<WebPageCreationParameters::SubframeProcessPageParameters>&&` instead.

Reply:
I do want all functions to take in all the parameters they need and return an object that needs no further modification.  Having a creationParametersForProvisionalPage and creationParametersForRemotePage and maybe another wrapper would be useful, though.  Also, all parameters like "true" should be given a name.

Reply:
A compromise! Much appreciated.